lib/node_modules/@stdlib/fs/resolve-parent-paths/lib/validate.js

```diff
@@ -23,17 +23,13 @@
 var isObject = require( '@stdlib/assert/is-plain-object' );
 var hasOwnProp = require( '@stdlib/assert/has-own-property' );
 var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var contains = require( '@stdlib/array/base/assert/contains' ).factory;
 var format = require( '@stdlib/string/format' );
 
 
 // VARIABLES //
 
-var isMode = {
-	'first': true,
-	'some': true,
-	'all': true,
-	'each': true
-};
+var isMode = contains( [ 'first', 'some', 'all', 'each' ] );
```

Comment:
Just curious, wouldn't this be O(N), and with the object O(1)?

Reply:
`hasOwnProp` is relatively expensive. For `4` items, you're not likely to see any practical difference, even if `hasOwnProp` were cheap.

If you want, you can write a benchmark script to test the hypothesis.

Reply:
Interesting, yeah you were right:

![image](https://github.com/user-attachments/assets/97c8958d-4b1a-45b4-841e-d7aae9c52516)

Reply:
Theory vs practice. Makes sense if one thinks about how object property access may be implemented (without inline caching), as one needs to find the corresponding key, which is likely O(N).